client/checkout/api/index.js

```diff
@@ -255,6 +287,10 @@ export default class WCPayAPI {
 							throw result.error;
 						}
 
+						if ( paymentError ) {
+							throw paymentError;
+						}
+
```

Comment:
This error is thrown after the ajax call to ensure the order status updates for failed orders.